### PR TITLE
[Ready] [Semi-Modular] Roundstart Snails

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
(As a Veteran race.) 
This fixes up several things about the snail race to be a bit more indepth, and get them ready for actual playability.
One, this makes their snail crawl deploy water instead of lubricant. It lays less of it because it is slippery and we don't want an entire lane of hallway to be inaccessible.

Next, they're added to scangates and the nanite sensor program. Customary.

I've increased their customization traits, including hair and shirts and such, eye color changing, and I've given them some food prefs after some light research into real life land snail food prefs. They fucking hate citrus.

Their punch damage has been increased to a maximum of five, and their blood has been changed to lizard type blood. I gave them a unique liver which can shrug off 2 more units of toxins than a normal human, and I've increased the size of their backpack to be equivalent to a duffelbag. They're certainly paying a duffelbag of slowdown for it and then some. Lastly, they no longer have bones, but do still take flesh wounds.

Lastly, snails are now able to breathe underwater. Did you know it takes a full day for a land snail to really drown?

## How This Contributes To The Skyrat Roleplay Experience
I think that snailpeople, with some work, can add a lot to the game as a playable species. We don't really have invertebrate species, and the slow and plodding playstyle of a snailperson could certainly offer something different than our usual high-flying crewmembers zooming around and such. Them washing the floor will hopefully start a rise of adorable little cleaner snails crawling around your department jamming out to Pixel Galaxy.
Also, I mean. Come on. Look at this shit.
image
That has to tug on at least one heartstring. That is a polite boy.

Before you ask about the Veterancy thing, it was planned in the last PR. Special thank you to @jjpark-kb for the groundwork of this, and for giving me the blessing to invite our gastropod overlords here.

It was requested I talk about their downsides, so here we go.
They're six times slower than a normal person (if my math is right,) and can only achieve human-esque speed by crawling along the floor. Snails are twice as vulnerable to electric shocks, take twice the burn damage (even more than any other race afaik,), take a bunch of damage from salt (5u is enough to deal 26 burn damage to them in a very short timeframe) and they physically cannot cross a salt pile no matter how hard they try. Also, only do a max of five punch damage.

## Changelog
:cl: Nerevar & Jake
add: Adds snails as a roundstart, veteran-only race.
balance: Snails have more punch damage, a slightly stronger liver, and more space in their backpack. They no longer take bone wounds, being invertebrates. 
fix: Fixes snail shells being invisible on them. 
balance: Snails have a more robust list of food preferences. They hate sugar, citrus and milk, but are enjoyers of vegetables, fruits, and things we'd generally consider kinda gross.
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
If anyone in this comment section has any earthly idea as to how to make snail eyes layer over their hair, PLEASE help.

**Comment by linnpap** [comment]
GOATED PR

**Comment by theselfish** [comment]
image

File: code/datums/elements/snail_crawl.dm [hunk 1]
```diff
@@ -31,5 +31,6 @@
 
 	var/turf/open/OT = get_turf(snail)
 	if(istype(OT))
-		OT.MakeSlippery(TURF_WET_LUBE, 20)
+		OT.MakeSlippery(TURF_WET_WATER, 1.5 SECONDS) //SKYRAT EDIT: Roundstart Snails - No more lube
+		OT.wash(CLEAN_WASH)
```
Comment (Gandalf2k15) [inline code/datums/elements/snail_crawl.dm]: I wouldn't make them wash. Too many upsides.

Comment (Nerev4r) [reply code/datums/elements/snail_crawl.dm]: Makes sense.

**Comment by Nerev4r** [comment]
small update: eyestalks now layer over hair and it's SO CUTE. 
image
image
i've taken away the washing properties from their crawl.
this can be safely reviewed now.

**Comment by Nerev4r** [comment]
Whitescaled their eyes so they're easier to recolor. The old default color was 3C3021 if you're curious.

**Comment by Nerev4r** [comment]
image
Special thanks to @CandleJaxx, snails now have a grand total of one other shell to choose from! A more conical flavor for a more conical crewmember.
We're probably done here tho.

File: code/datums/elements/snail_crawl.dm [hunk 2]
```diff
@@ -31,5 +31,5 @@
 
 	var/turf/open/OT = get_turf(snail)
 	if(istype(OT))
-		OT.MakeSlippery(TURF_WET_LUBE, 20)
```
Comment (jjpark-kb) [inline code/datums/elements/snail_crawl.dm:34]: make the variable not OT-- make it like open_turf (or something like that)

File: code/datums/elements/snail_crawl.dm [hunk 3]
```diff
@@ -29,7 +29,7 @@
 /datum/element/snailcrawl/proc/lubricate(atom/movable/snail)
 	SIGNAL_HANDLER
 
-	var/turf/open/OT = get_turf(snail)
-	if(istype(OT))
-		OT.MakeSlippery(TURF_WET_LUBE, 20)
+	var/turf/open/open_turf = get_turf(snail)
```
Comment (Zonespace27) [inline code/datums/elements/snail_crawl.dm]: don't change the two letter var, less nonmodular code is better

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 4]
```diff
@@ -3,26 +3,31 @@
 	id = SPECIES_SNAIL
 	offset_features = list(OFFSET_GLASSES = list(0,4))
 	default_color = "336600" //vomit green
-	species_traits = list(MUTCOLORS, NO_UNDERWEAR, HAS_FLESH, HAS_BONE)
+	species_traits = list(MUTCOLORS,EYECOLOR,HAS_FLESH,HAS_BONE,HAIR,FACEHAIR) //SKYRAT EDIT - Roundstart Snails - Customization
 	inherent_traits = list(
 		TRAIT_ADVANCEDTOOLUSER,
 		TRAIT_CAN_STRIP,
 		TRAIT_NOSLIPALL,
+		TRAIT_WATER_BREATHING,
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm]: skyrat edit

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 5]
```diff
@@ -3,26 +3,31 @@
...
+	liked_food = VEGETABLES | FRUIT | GROSS | RAW //SKYRAT EDIT - Roundstart Snails - Food Prefs
+	disliked_food = PINEAPPLE | DAIRY | ORANGES | SUGAR //SKYRAT EDIT: Roundstart Snails - As it turns out, you can't give a snail processed sugar or citrus.
 	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | RACE_SWAP
 	sexes = FALSE //snails are hermaphrodites
+	veteran_only = TRUE //SKYRAT EDIT - Roundstart Snails
 
 	mutanteyes = /obj/item/organ/eyes/snail
 	mutanttongue = /obj/item/organ/tongue/snail
-	exotic_blood = /datum/reagent/lube
+	mutantliver = /obj/item/organ/liver/snail
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm]: skyrat edit

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 6]
```diff
@@ -77,3 +105,10 @@
 /obj/item/storage/backpack/snail/Initialize(mapload)
 	. = ..()
 	ADD_TRAIT(src, TRAIT_NODROP, "snailshell")
+
+//SKYRAT EDIT BEGIN - Roundstart Snails
+/obj/item/storage/backpack/snail/ComponentInitialize()
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm]: put this in a modular file

File: code/modules/surgery/organs/eyes.dm [hunk 7]
```diff
@@ -35,6 +35,7 @@
 	/// indication that the eyes are undergoing some negative effect
 	var/damaged = FALSE
 	var/is_emissive = FALSE //SKYRAT EDIT ADDITION
+	var/eyes_layer = BODY_LAYER
```
Comment (Zonespace27) [inline code/modules/surgery/organs/eyes.dm]: skyrat edit

File: code/modules/surgery/organs/eyes.dm [hunk 8]
```diff
@@ -456,6 +457,7 @@
 	desc = "These eyes seem to have a large range, but might be cumbersome with glasses."
 	eye_icon_state = "snail_eyes"
 	icon_state = "snail_eyeballs"
+	eyes_layer = ABOVE_BODY_FRONT_HEAD_LAYER
```
Comment (Zonespace27) [inline code/modules/surgery/organs/eyes.dm]: skyrat edit

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by Nerev4r** [comment]
image
image
Two more shells, by Zydras.
Also, please for the love of God someone help with this runtime it's emotionally killing the PR author.

**Comment by Nerev4r** [comment]
hey it finally passes CI pls thank my good pal Fikou for offhandedly mentioning the check_available check

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 9]
```diff
@@ -3,26 +3,31 @@
...
+		TRAIT_WATER_BREATHING, //SKYRAT EDIT - Roundstart Snails
 	)
 	attack_verb = "slap"
 	attack_effect = ATTACK_EFFECT_DISARM
 	say_mod = "slurs"
 	coldmod = 0.5 //snails only come out when its cold and wet
 	burnmod = 2
 	speedmod = 6
-	punchdamagehigh = 0.5 //snails are soft and squishy
+	punchdamagehigh = 5 //snails are soft and squishy //SKYRAT EDIT - Roundstart Snails - A Bit More Damage
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm:20]: no. this makes them deal a massive (2.5x this number) of stamina damage per punch.

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 10]
```diff
@@ -3,26 +3,31 @@
...
 	siemens_coeff = 2 //snails are mostly water
+	liked_food = VEGETABLES | FRUIT | GROSS | RAW //SKYRAT EDIT - Roundstart Snails - Food Prefs
+	disliked_food = PINEAPPLE | DAIRY | ORANGES | SUGAR //SKYRAT EDIT: Roundstart Snails - As it turns out, you can't give a snail processed sugar or citrus.
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm]: Leave pineapple affinity to the trait

File: code/modules/mob/living/carbon/human/species_types/snail.dm [hunk 11]
```diff
@@ -3,26 +3,31 @@
...
 	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | RACE_SWAP
 	sexes = FALSE //snails are hermaphrodites
+	veteran_only = TRUE //SKYRAT EDIT - Roundstart Snails
 
 	mutanteyes = /obj/item/organ/eyes/snail
 	mutanttongue = /obj/item/organ/tongue/snail
-	exotic_blood = /datum/reagent/lube
+	mutantliver = /obj/item/organ/liver/snail //SKYRAT EDIT - Roundstart Snails
+	exotic_bloodtype = "L" //SKYRAT EDIT: Roundstart Snails - No more lube
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species_types/snail.dm]: I kinda disagree with giving them lizarb blood but eh

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/snail.dm (hunk 9 above)
Comment (Nerev4r) [reply code/modules/mob/living/carbon/human/species_types/snail.dm:20]: misunderstanding cleared up.

File: code/modules/mob/living/carbon/human/species_types/snail.dm (hunk 10 above)
Comment (Nerev4r) [reply code/modules/mob/living/carbon/human/species_types/snail.dm]: okee.

File: code/modules/mob/living/carbon/human/species_types/snail.dm (hunk 11 above)
Comment (Nerev4r) [reply code/modules/mob/living/carbon/human/species_types/snail.dm]: yeah that works.

**Review by Zonespace27 — APPROVED** [review]

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**